ambry-replication/src/main/java/com.github.ambry.replication/ReplicaThread.java

```diff
@@ -550,29 +579,29 @@ private void processReplicaMetadataResponse(Set<StoreKey> missingStoreKeys,
           // A Repair event for Delete signifies that a Delete message was received from the remote and it is fired
           // as long as the Delete is guaranteed to have taken effect locally.
           if (notification != null) {
-            notification.onBlobReplicaDeleted(dataNodeId.getHostname(), dataNodeId.getPort(),
-                messageInfo.getStoreKey().getID(), BlobReplicaSourceType.REPAIRED);
+            notification.onBlobReplicaDeleted(dataNodeId.getHostname(), dataNodeId.getPort(), localKey.getID(),
+                BlobReplicaSourceType.REPAIRED);
           }
         }
       } else {
         if (messageInfo.isDeleted()) {
           // if the key is not present locally and if the remote replica has the message in deleted state,
           // it is not considered missing locally.
-          missingStoreKeys.remove(messageInfo.getStoreKey());
+          missingRemoteStoreKeys.remove(messageInfo.getStoreKey());
           logger.trace("Remote node: {} Thread name: {} Remote replica: {} Key in deleted state remotely: {}",
               remoteNode, threadName, remoteReplicaInfo.getReplicaId(), messageInfo.getStoreKey());
           // A Repair event for Delete signifies that a Delete message was received from the remote and it is fired
           // as long as the Delete is guaranteed to have taken effect locally.
           if (notification != null) {
-            notification.onBlobReplicaDeleted(dataNodeId.getHostname(), dataNodeId.getPort(),
-                messageInfo.getStoreKey().getID(), BlobReplicaSourceType.REPAIRED);
+            notification.onBlobReplicaDeleted(dataNodeId.getHostname(), dataNodeId.getPort(), localKey.getID(),
+                BlobReplicaSourceType.REPAIRED);
           }
         } else if (messageInfo.isExpired()) {
           // if the key is not present locally and if the remote replica has the key as expired,
           // it is not considered missing locally.
-          missingStoreKeys.remove(messageInfo.getStoreKey());
+          missingRemoteStoreKeys.remove(messageInfo.getStoreKey());
           logger.trace("Remote node: {} Thread name: {} Remote replica: {} Key in expired state remotely {}",
```

Comment:
minor: this log message uses the `localKey` and the previous one (for delete) uses the remote key - let's use the same (local or remote) in both messages

Reply:
added local key to the logs along with the remote key. notifications I think all use the local key.